sdks/python/apache_beam/transforms/environments.py

```diff
@@ -337,18 +399,21 @@ def __init__(
       env=None,  # type: Optional[Mapping[str, str]]
       capabilities=(),  # type: Iterable[str]
       artifacts=(),  # type: Iterable[beam_runner_api_pb2.ArtifactInformation]
+      resource_hints=None,  # type: Optional[Mapping[str, bytes]]
   ):
     # type: (...) -> None
-    super(ProcessEnvironment, self).__init__(capabilities, artifacts)
+    super(ProcessEnvironment,
+          self).__init__(capabilities, artifacts, resource_hints)
     self.command = command
     self.os = os
     self.arch = arch
     self.env = env or {}
 
   def __eq__(self, other):
-    return self.__class__ == other.__class__ \
-      and self.command == other.command and self.os == other.os \
-      and self.arch == other.arch and self.env == other.env
+    return (
+        super().__eq__(other) and self.command == other.command and
```

Comment:
(This is starting to feel very redundant. I wonder if this could be refactored in such a way that it wasn't so redundant...like maybe one could have an EnvironmentType property that abstracts away the different types of environments, and Environment could handle all the common stuff... maybe out of scope for now.)

Reply:
+1 to consider this out of scope for now.